torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -107,8 +108,11 @@ void GpuLower::lower() {
   const auto unrolled_loops =
       UnrollPass::runPass(fusion_, lowered_exprs, preds);
 
+  // Insert SyncThreads at end of for-loop to avoid WAR race condition
+  const auto sync_exprs = insertThreadSynchronization(fusion_, unrolled_loops);
```

Comment:
Thanks. I realize in this instance it seems a bit silly to pass through the expressions, but as we move towards a more formal pipeline this will be very important.